hathor/manager.py

```diff
@@ -441,13 +441,15 @@ def _initialize_components(self) -> None:
         # self.start_profiler()
         if self._full_verification:
             self.log.debug('reset all metadata')
-            for tx in self.tx_storage.get_all_transactions():
-                tx.reset_metadata()
+            with self.tx_storage.allow_partially_validated_context():
+                for tx in self.tx_storage.get_all_transactions():
+                    tx.reset_metadata()
 
         self.log.debug('load blocks and transactions')
         for tx in self.tx_storage._topological_sort_dfs():
             if self._full_verification:
-                tx.update_initial_metadata()
+                with self.tx_storage.allow_partially_validated_context():
+                    tx.update_initial_metadata()
```

Comment:
Why would this access need a broader scope?

Reply:
Because `tx.update_initial_metadata()` does a `tx_storage.save_transaction(tx)` but we haven't validated `tx` yet. I don't like too much that some methods will implicitly save, but a lot of times it's hard to get around that. I think using `allow_partially_validated_context()` is correct here.

Reply:
I agree. Thanks for the explanation.